src/cargo/util/toml/mod.rs

```diff
@@ -1944,13 +1939,18 @@ fn detailed_dep_to_dependency<P: ResolveToPath + Clone>(
     }
 
     if let Some(p) = orig.public {
-        cx.features.require(Feature::public_dependency())?;
-
         if dep.kind() != DepKind::Normal {
             bail!("'public' specifier can only be used on regular dependencies, not {:?} dependencies", dep.kind());
```

Comment:
Generally, this should be a warning and not an error when the unstable feature is inactive.  Unsure if its worth it though in this case.